src/menus/icon-menu.jsx

```diff
@@ -83,8 +92,23 @@ const IconMenu = React.createClass({
   //to update theme inside state whenever a new theme is passed down
   //from the parent / owner using context
   componentWillReceiveProps(nextProps, nextContext) {
+    if (process.env.NODE_ENV !== 'production') {
+      this._warningIfNeeded();
+    }
+
     let newMuiTheme = nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme;
     this.setState({muiTheme: newMuiTheme});
+
+    if (nextProps.open === true || nextProps.open === false) {
+      this.setState({open: nextProps.open});
+    }
+  },
+
+  _warningIfNeeded() {
+    if (this.props.hasOwnProperty('open')) {
+      warning(this.props.hasOwnProperty('closeOnItemTouchTap'),
+        'closeOnItemTouchTap has been deprecated in favor of open, onRequestChange');
```

Comment:
What do you think about using something like https://github.com/react-bootstrap/react-prop-types/blob/master/README.md#deprecatedproptype-explanation instead?